### PR TITLE
[API] Organizer can set group category

### PR DESCRIPTION
**Pull request by Jon1969Edwards**
## Description
As an Organizer
In order to make my group easier to find
I would like to be able to set a Category for my group

## PivotalTracker
[Link](https://www.pivotaltracker.com/story/show/163682227)

## Tasks
- Write request spec
- Create Category model with #index
- Create Endpoint
- Associate with Group

### CODE REVIEW

File: spec/requests/organizer_can_set_group_category_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+describe 'GET /cagegories/:id/groups' do
+  let!(:category_1) { create(:category) } 
+  let!(:category_2) { create(:category) } 
+
```
Comment (CamronLDNF) [inline spec/requests/organizer_can_set_group_category_spec.rb:6]: 1) What is 'cagegories'? :)  correct the spelling

2) Shouldn't it be just 'GET /categories/:id'?  Compare to the request spec groups

File: spec/requests/user_can_create_group_spec.rb [hunk 2]
```diff
@@ -3,12 +3,14 @@
...
+  let(:category) { create(:category) }
+
   describe 'POST req with valid credentials' do
     let(:user_credentials) { user.create_new_auth_token }
     let(:headers) { { HTTP_ACCEPT: 'application/json' }.merge!(user_credentials) }
     
     before do
-      post "/groups", params: { group: {name: 'coding'} }, headers: headers
+      post "/groups", params: { group: {name: 'coding', category_id: category.id } }, headers: headers
     end
```
Comment (CamronLDNF) [inline spec/requests/user_can_create_group_spec.rb:14]: Think you can just do `category_id: category`

**Review by CamronLDNF — COMMENTED** [review]
Left 2 comments

File: spec/requests/organizer_can_set_group_category_spec.rb (hunk 1 above)
Comment (Jon1969Edwards) [reply spec/requests/organizer_can_set_group_category_spec.rb:6]: These have both been fixed.

File: spec/requests/user_can_create_group_spec.rb (hunk 2 above)
Comment (CamronLDNF) [reply spec/requests/user_can_create_group_spec.rb:14]: Noted that it in fact needs to be as stated.

**Pull request merged**